### ISSUE TITLE
Table: Allow drop after first thought in col1

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#1885]
## Steps to Reproduce

```
- a
  - =view
    - Table
  - =pin
    - true
  - b
    - c
  - d
    - e
- x
```

## Current Behavior

Cannot drop `x` after `b`.

## Expected Behavior

Should be able to drop `x` after `b`.

DropHover should be rendered in the correct place.

**Comment by Zubair286** [comment]
I am able to drop x after b, albeit the drop-hover is not really rendering as it should, as shown in the vdieo,
[Screencast from 07-12-2024 02:03:46 PM.webm](https://github.com/user-attachments/assets/87db8861-391a-4c3a-a7e8-d346f90ade20)

**Comment by Zubair286** [comment]
[Screencast from 07-12-2024 02:03:46 PM.webm](https://github.com/user-attachments/assets/d4ae6a61-7a8b-4df8-82fa-05686985e142)

**Comment by Zubair286** [comment]
I am able to drop x after b, albeit the DropHover is not really rendering as it should. It's width is too short.
[Screencast from 07-12-2024 02:03:46 PM.webm](https://github.com/user-attachments/assets/a711a991-7066-44b6-8cdb-0711c644287f)

**Comment by Zubair286** [comment]
[Screencast from 07-12-2024 02:03:46 PM.webm](https://github.com/user-attachments/assets/489aeeaa-943d-4225-acf1-fc664e479759)

**Comment by raineorshine** [comment]
Thanks, I updated the issue text to better describe the problem.

**Comment by Zubair286** [comment]
What should be the correct position for this `DropHover`, I am guessing its left edge be flush with col1 thought and what should be its width?, right now its taking the width of col1 thought, and on `drop-hover` class because of this `width: calc(100% - 2em),` the width is appearing odd. Should it span across the container like other DropHovers?

**Comment by raineorshine** [comment]
The left edge should be flush with the leftmost col1 thought:

<img width="226" src="https://github.com/user-attachments/assets/bbbf6499-df3e-4403-8200-e9848ad647da">

The width of the DropHover bar in this case should be about the full width of the screen. 

However, it should be shorter than the DropHover bar that is rendered when dragging into col2. That is, the right edge of the col1 DropHover bar should be at least 1em or 2em to the left of the right edge of the col2 DropHover bar:

<img width="702" src="https://github.com/user-attachments/assets/df281ef1-8446-4758-aa45-a04b0af351aa">

**Comment by Zubair286** [comment]
Right, in that case we might need to move the `DropHover` from where its currently rendering right now, in the `Thought` component, or manipulate it in such a way that it isn't bound by the width of col1.

**Comment by raineorshine** [comment]
Overflowing the width is one option, even with something as simple as `vw` to avoid having to do any runtime measurements.

Moving the `DropHover` outside the `Thought` component is another option. That involves a change to the component hierarchy, but it might be easier to control the width.

I'm interested to see what ends up being the most elegant solution.

**Comment by Zubair286** [comment]
I think setting the width to around `50vw` and adjusting the margin a little to the right should do the trick, it'd be much easier than changing the component hierarchy for this.
